### ISSUE TITLE
Headings should not be labeled by the author

### ISSUE THREAD

**Issue by JAWS-test** [w3c/aria#2797]
# Description of bug or feature request

According to the ARIA specification, headings can be labeled using `aria-label` and `aria-labelledby`. All other text elements (generic, paragraph, mark, deletion, code, etc.; see https://w3c.github.io/aria/#namefromprohibited) cannot be labeled, which is correct because only their content matters. The same should apply to headings. 

If a heading has a label from the author, it is either completely ignored (JAWS, https://github.com/FreedomScientific/standards-support/issues/777) or causes extremely serious errors with the screen reader (NVDA): nested elements are neither detectable nor operable.

If the ARIA specification continues to allow “name from author,” it is absolutely essential and urgent to clarify that a heading should be treated like a landmark region. When the user enters the region, the “name from author” is output, but the content of the heading remains fully perceivable and operable. I would be against such a solution because it is nonsensical, but it would be better than the current situation. And why is this urgent? Because @scottaohara reported this serious issue back in 2022 (https://github.com/nvaccess/nvda/issues/13296, https://github.com/w3c/aria/issues/1800), and nothing has happened in the last four years!

## Will this require a change to [CORE-AAM](https://w3c.github.io/core-aam/)?

No

## Will this require a change to the [ARIA authoring guide](https://w3c.github.io/aria-practices/)?

Yes (chapter [Accessible Name Guidance by Role](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#accessiblenameguidancebyrole))

**Comment by pkra** [comment]
This is effectively a duplicate of #1800. There have also been related issues since then. In particular #2569 was marked as F2F candidate for TPAC.

I don't think opening one more issue along the same lines will really help.

**Comment by JAWS-test** [comment]
@pkra In my opinion, this is not a duplicate. These are two related, but not identical, questions:
* The linked tickets address the general question of how the “Name from author” should function. This is a discussion that has been going on for years and cannot be answered simply or in a one-size-fits-all manner. It involves all elements, and finding a good solution will take a lot of time, which I understand.
* My ticket concerns a specific and obvious error that occurs in headings and has serious implications for accessibility. This error should be fixed  as soon as possible. After four years of inaction, letting even more years go by before this serious problem is resolved is not a solution.

**Comment by pkra** [comment]
@JAWS-test I understand your position. I personally don't think it's a good idea to make this single (breaking) change without identifying a path for tackling the wider issue. I fear this would only introduce more problems down the line.
